ext/ITensorNetworksEinExprsExt/src/ITensorNetworksEinExprsExt.jl

```diff
@@ -0,0 +1,58 @@
+module ITensorNetworksEinExprsExt
+
+using ITensors: Index, ITensor, Algorithm
+using ITensorNetworks:
+  ITensorNetworks,
+  ITensorNetwork,
+  vertextype,
+  vertex_data,
+  externalinds,
+  contraction_sequence
+using EinExprs: EinExprs, EinExpr, einexpr, SizedEinExpr
+
+function prepare_einexpr(tn::ITensorNetwork)
+  IndexType = Any
+  VertexType = vertextype(tn)
+
+  tensor_exprs = EinExpr{IndexType}[]
+  tensor_inds_to_vertex = Dict{Set{IndexType},VertexType}()
+  inds_dims = Dict{IndexType,Int}()
+
+  for v in vertices(tn)
+    tensor_v = tn[v]
+    inds_v = collect(inds(tensor_v))
+    push!(tensor_exprs, EinExpr{IndexType}(; head=inds_v))
+    tensor_inds_to_vertex[Set(inds_v)] = key
+    merge!(inds_dims, Dict(inds_v .=> size(tensor_v)))
+  end
+
+  externalinds_tn = collect(externalinds(tn))
+  expr = SizedEinExpr(sum(tensor_exprs; skip=externalinds_tn), inds_dims)
+
+  return expr, tensor_inds_to_vertex
+end
+
+function EinExprs.einexpr(tn::ITensorNetwork; optimizer::EinExprs.Optimizer)
+  expr, _ = prepare_einexpr(tn)
+  return einexpr(optimizer, expr)
+end
+
+function ITensorNetworks.contraction_sequence(
+  alg::Algorithm"einexpr", tn::Vector{ITensor}; kwargs...
+)
+  return contraction_sequence(alg, ITensorNetwork(tn); kwargs...)
+end
+
+function ITensorNetworks.contraction_sequence(
+  ::Algorithm"einexpr", tn::ITensorNetwork{T}; optimizer=EinExprs.Exhaustive()
+)
+  path, tensor_inds_to_vertex = prepare_einexpr(tn)
+  return to_contraction_sequence(tensor_inds_to_vertex, path)
+end
+
+function to_contraction_sequence(tensor_inds_to_vertex, path)
```

Comment:
```suggestion
function to_contraction_sequence(expr, tensor_inds_to_vertex)
```
That ordering makes more sense to me, since this function is converting the `expr`. The `tensor_inds_to_vertex` is just used to help with that conversion.